torch/onnx/_internal/fx/passes/decomp.py

```diff
@@ -47,17 +50,30 @@ def _run(self, *args, **kwargs) -> torch.fx.GraphModule:
         #  mul = sym_size_2 * sym_size_3;  sym_size_2 = sym_size_3 = None
         #  view: f32[3, 5, 20] = torch.ops.aten.view.default(x, [sym_size, sym_size_1, mul])
 
-        fx_mode = "symbolic" if self.enable_dynamic_axes else "fake"
+        # Mimic `torch._dynamo.export(aten_graph=True)` behavior in invoking `make_fx`.
+        # TODO: May need revisit for user fake mode export + dynamic shape scenario.
+        fake_mode: Optional[fake_tensor.FakeTensorMode] = self.fake_mode
+        maybe_fake_args = self._maybe_fakefy_args(fake_mode, *args)
+        if fake_mode is not None:
+            # Using existing fake mode as context, signal `make_fx` that it does not need
+            # to create a new fake mode by passing tracing_mode as "real".
+            tracing_mode = "real"
+        else:
+            # Existing fake mode not found, signal `make_fx` to create one.
+            fake_mode = contextlib.nullcontext()  # type: ignore[assignment]
+            tracing_mode = "symbolic" if self.enable_dynamic_axes else "fake"
 
         # Apply decomposition table to the input graph.
-        # Make sure the feed-in "module" is stateless.
-        decomposed_module = proxy_tensor.make_fx(
-            module,
-            decomposition_table=self.decomposition_table,
-            tracing_mode=fx_mode,
-            _allow_non_fake_inputs=True,
-            _allow_fake_constant=self.allow_fake_constant,
-        )(*args)
+        assert fake_mode is not None
+        with python_dispatch.enable_python_dispatcher(), fake_mode:
```

Comment:
out of curiosity, which scenario requires the python dispatcher to be explicitly enabled for the pass (and not for the actual call to `torch._dynamo.export`)?

My understanding it is already enabled by default, so this would be needed if the user explicitly disables it, right? In that scenario, should we re-enable it?

Reply:
Great question. It is disabled by default. Selectively turned on when one wishes to enable the python dispatcher key in the dispatcher key stack. From export user perspective this should be super low level detail and not exposed.

Reply:
I didn't mean visible to the user, but something like

```python

class Exporter:

    def export():
        with python_dispatch.enable_python_dispatcher():
            pre_export_passes()
            torch._dynamo.export()
```

Reply:
That is, why we are internally enabling the dispatcher only for a few passes while not to the `fx_tracer.generate_fx()` itself. There is usually a "sync" between `make_fx` and `torch._dynamo.export` calls to keep their fake_modeness identical

Reply:
Short answer: this matches with aten_graph=True behavior.
Longer answer: `torch._dynamo.export` manages `enable_python_dispatcher` internally. `make_fx` on the other hand can be viewed as a lower level api. While It does have convenience argument `tracing_mode` that sets up fake mode and related configurations inside, one should construct and re-use existing contexts from `torch._dynamo.export` in order to achieve the synchronization regarding fake mode and friends.